### PR TITLE
ISPN-11370 Server Test Driver

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-11370

Extracts the server test driver into a standalone module that can be used by 3rd parties

I have created a repository at https://github.com/tristantarrant/infinispan-playground-testdriver which demonstrates how to use this in other projects. I will promote it to the Infinispan organization once this is in.

### CODE REVIEW

**Comment by tristantarrant** [comment]
@diegolovison @gustavolira

**Comment by diegolovison** [comment]
I am working on https://github.com/infinispan/infinispan/pull/7912 and then once I get it working, I will backport and use your PR as a reference

**Comment by tristantarrant** [comment]
I've created a small demo project that shows how to use it:
https://github.com/tristantarrant/infinispan-playground-testdriver

**Comment by diegolovison** [comment]
@tristantarrant long term, do you think that we can replace the code base in:
`client/hotrod-client`
`client/infinispan-key-value-store-hotrod`
`client/marshaller`
`client/rest-client`
and most of the subprojects in `integrationtests` with the new approach?

**Comment by tristantarrant** [comment]
Ideally, yes

**Comment by gustavolira** [comment]
@tristantarrant I created a demo using infinispan-server-testdriver + arquillian. infinispan-server-testdriver currently is not compatible with arquillian. https://github.com/gustavolira/JDG-3506
My demo is using arquillian to deploy into managed wildfly + IninfinispanServerRule running with container mode.
I tested your demo isolated and worked like a charm.

**Comment by karesti** [comment]
There are many changes related to moving the code around, but the rest seems good.
it is lacking docs on how to use this.
I would like to use this in my examples, quarkus demo specially

File: server/tests/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+Infinispan Server Integration Tests
```
Comment (karesti) [inline server/tests/README.md]: @tristantarrant I found it!

**Comment by tristantarrant** [comment]
@diegolovison @karesti https://github.com/tristantarrant/infinispan-playground-testdriver

**Comment by tristantarrant** [comment]
Still 3000 tests missing. I'll check

**Comment by tristantarrant** [comment]
Looking good now

**Comment by ryanemerson** [comment]
I managed to wip together a very basic integration test for the image, this is really cool :slightly_smiling_face: 

> hm no, we need the ascii docs for how to use this out of the box.

+1 to ascii docs. This is a very useful tool, we should promote this to being part of our official documentation instead of just README.md files in the modules.

**Comment by tristantarrant** [comment]
I'll write docs later

**Comment by tristantarrant** [comment]
As this fixes several issues with the container driver, I'm going to backport it wholesale to 10.1.x asap too

**Comment by karesti** [comment]
integrated with joy, thanks!!

**Pull request merged**